integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java

```diff
@@ -478,18 +479,41 @@ private int renameInternal(String sourcePath, String destPath) {
           sourcePath, destPath, name, MAX_NAME_LENGTH);
       return -ErrorCodes.ENAMETOOLONG();
     }
-    Optional<URIStatus> status = AlluxioFuseUtils.getPathStatus(mFileSystem, sourceUri);
-    if (!status.isPresent()) {
+    Optional<URIStatus> sourceStatus = AlluxioFuseUtils.getPathStatus(mFileSystem, sourceUri);
+    if (!sourceStatus.isPresent()) {
       LOG.error("Failed to rename {} to {}: source non-existing", sourcePath, destPath);
       return -ErrorCodes.EEXIST();
     }
-    if (!status.get().isCompleted()) {
+    if (!sourceStatus.get().isCompleted()) {
       // TODO(lu) https://github.com/Alluxio/alluxio/issues/14854
       // how to support rename while writing
       LOG.error("Failed to rename {} to {}: source is incomplete", sourcePath, destPath);
       return -ErrorCodes.EIO();
     }
+    Optional<URIStatus> destStatus = AlluxioFuseUtils.getPathStatus(mFileSystem, destUri);
     try {
+      if (destStatus.isPresent()) {
+        if (AlluxioJniRenameUtils.exchange(flags)) {
+          LOG.error("Failed to rename {} to {}, not support exchange flags",
```

Comment:
Failed to rename {} to {}: destination file/dir must exist to RENAME_EXCHANGE

Reply:
Here it is already indicated that the file exists.

```
if (destStatus.isPresent()){
}
```